src/Vault.sol

```diff
@@ -311,10 +313,14 @@ contract Vault {
      * or the smaller the on-chain liquidity, the biggert the haircut will be.
      */
     function getCollateralValue(uint256 vaultValue) public view returns (uint256 collateralValue) {
```

Comment:
This function is obsolete now, the only difference with the function above is that vaultValue is already known before the collateralValue is calculated (big gas saving). But since the collateralFactor is here again calculated in real time, we again have to loop and get the value of each asset